cloudflare/resource_cloudflare_worker_secret_test.go

```diff
@@ -0,0 +1,165 @@
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	cleanhttp "github.com/hashicorp/go-cleanhttp"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/cloudflare/terraform-provider-cloudflare/version"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/logging"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/httpclient"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+const scriptContentForSecret = `addEventListener('fetch', event => {event.respondWith(new Response('test 1'))});`
+
+var workerSecretTestScriptName string
+
+func TestAccCloudflareWorkerSecret_Basic(t *testing.T) {
+	t.Parallel()
+
+	name := generateRandomResourceName()
+	secretText := generateRandomResourceName()
+	workerSecretTestScriptName = generateRandomResourceName()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheckAccount(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudflareWorkerSecretDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckCloudflareWorkerSecretWithWorkerScript(workerSecretTestScriptName, name, secretText),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudflareWorkerSecretExists(workerSecretTestScriptName, name, secretText),
+				),
+			},
+		},
+	})
+
+	deleteWorker()
```

Comment:
we shouldn't need to explicitly call `deleteWorker` here as the test harness calls the `Delete` operation on our behalf.

Reply:
From my understanding, that test case shouldn't require to be manually configuring this like this - even if they are mutually exclusive as they need to be able to co-exist in configuration together. Can you put up a small reproduction case to show the issues you had that lead here and we can see what we can do about it.